pep-0613.rst

```diff
@@ -0,0 +1,173 @@
+
+Forward References:
+*******************
+
+::
+
+  MyType = "ClassName"
+  def foo() -> MyType: ...
+
+This code snippet should not error so long as :code:`ClassName` is defined later on. However, a type checker is forced to throw spurious errors that (1) :code:`MyType` is an unannotated global string, and (2) :code:`MyType` is not a valid type.
+
+::
+
+  MyType: TypeAlias = “ClassName”
+  def foo() -> MyType: ...
+
+Explicit aliases remove ambiguity so neither of the above errors will be thrown. Additionally, if something is wrong with :code:`ClassName` (i.e., it’s not actually defined later), the type checker can throw an error.
+
+
+Error Messaging:
+****************
+
+::
+
+  MyType = InvalidType
+  MyType = MyGeneric(int)  # i.e., intention was MyGeneric[int]
+
+A type checker should warn on this code snippet that :code:`InvalidType` is not a valid type, and therefore cannot be used to annotate an expression or to construct a type alias. Instead, type checkers are forced to throw spurious errors that (1) :code:`MyType` is a global expression missing an annotation, and (2) :code:`MyType` is not a valid type in all usages of :code:`MyType` across the codebase.
+
+::
+
+  MyType: TypeAlias = InvalidType
+  MyType: TypeAlias = MyGeneric(int)
+
+With explicit aliases, the type checker has enough information to error on the actual definition of the bad type alias, and explain why: that :code:`MyGeneric(int)` and `InvalidType` are not valid types. When the value expression is no longer evaluated as a global value, unactionable type errors on all usages of :code:`MyType` across the codebase can be suppressed.
+
+Scope Restrictions:
+*******************
+
+::
+
+  x = ClassName
+  def foo() -> None:
+    x = ClassName
+
+The outer :code:`x` is a valid type alias, but type checkers must error if the inner :code:`x` is ever used as a type. This is confusing because the alias declaration rule is not explicit, and the error is not thrown on the location of the inner type alias declaration but rather on every one of its subsequent use cases.
+
+::
+
+  x: TypeAlias = ClassName
+  def foo() -> None:
+    x = ClassName
+    x: TypeAlias = ClassName
+
+With explicit aliases, the outer assignment is still a valid type variable, and the inner assignment can either be a valid local variable or a clear error, communicating to the author that type aliases cannot be defined inside a nested scope.
+
+
+Specification
+=============
+
+The explicit alias declaration syntax attempts to clearly differentiate between the three possible kinds of assignments: typed global expressions, untyped global expressions, and type aliases. This avoids the existence of assignments that break type checking when an annotation is added, and avoids classifying the nature of the assignment based on the type of the value.
```

Comment:
Does it really only *attempt* to differentiate? I think you can just write "clearly differentiates". :-)